dashboard/modules/job/job_log_storage_client.py

```diff
@@ -13,45 +17,65 @@ class JobLogStorageClient:
     """
 
     # Number of last N lines to put in job message upon failure.
-    NUM_LOG_LINES_ON_ERROR = 10
-    # Maximum number of characters to print out of the logs to avoid
+    MAX_LOG_LINES_ON_ERROR = 10
+    # Max number of characters to print out of the logs to avoid
     # HUGE log outputs that bring down the api server
-    MAX_LOG_SIZE = 20000
+    MAX_LOG_SNIPPET_SIZE_ON_ERROR = 20000
+    # Max number of lines being fetched from log file per chunk (after
+    # every chunk event-loop has to be yielded to make sure that other tasks
+    # are not being starved)
+    MAX_LINES_PER_CHUNK = 10
 
-    def get_logs(self, job_id: str) -> str:
-        try:
-            with open(self.get_log_file_path(job_id), "r") as f:
-                return f.read()
-        except FileNotFoundError:
-            return ""
-
-    def tail_logs(self, job_id: str) -> Iterator[List[str]]:
-        return file_tail_iterator(self.get_log_file_path(job_id))
+    def tail_logs(
+        self,
+        job_id: str,
+        *,
+        max_lines_per_chunk: int = MAX_LINES_PER_CHUNK,
+    ) -> Iterator[List[str]]:
```

Comment:
it might make more sense to convert this iterator to `async` directly because this is called in other places where we similarly need to avoid blocking the loop